### PR TITLE
[TEST] Increasing test coverage on all non-SWT code.

### PR DESCRIPTION
**Pull request by mkambol**
@lucboudreau

### CODE REVIEW

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. mvn -f pom.xml clean install && mvn -f pom.xml site

Cleanup:
-------
mvn build-helper:remove-project-artifact

Changed files:
==============
  pom.xml
  src/main/java/org/pentaho/di/trans/steps/mongodb/MongoDbMeta.java
  src/main/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInput.java
  src/main/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputData.java
  src/main/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputMeta.java
  src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutput.java
  src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputData.java
  src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputDialog.java
  src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputMeta.java
  src/main/java/org/pentaho/di/ui/swing/preview/PreviewRowsSwingDialog.java
  src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/MongoDbInputDialog.java
  src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/MongoDbInputXulDialog.java
  src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/models/MongoDbModel.java
  src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/models/MongoDocumentField.java
  src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/models/MongoTag.java
  src/main/java/org/pentaho/mongo/wrapper/KettleMongoUtilLogger.java
  src/main/java/org/pentaho/mongo/wrapper/MongoWrapperUtil.java
  src/main/java/org/pentaho/mongo/wrapper/field/MongoArrayExpansion.java
  src/main/java/org/pentaho/mongo/wrapper/field/MongoField.java
  src/main/java/org/pentaho/mongo/wrapper/field/MongodbInputDiscoverFieldsImpl.java
  src/test/java/org/pentaho/di/trans/steps/mongodbinput/BaseMongoDbStepTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputDataTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputDiscoverFieldsHolderTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputMetaTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputDataTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputMetaTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputTest.java
  src/test/java/org/pentaho/mongo/wrapper/KettleMongoUtilLoggerTest.java
  src/test/java/org/pentaho/mongo/wrapper/MongoFieldTest.java
  src/test/java/org/pentaho/mongo/wrapper/MongoWrapperUtilTest.java
  src/test/java/org/pentaho/mongo/wrapper/field/MongodbInputDiscoverFieldsImplTest.java
  src/test/java/org/pentaho/reporting/ui/datasources/mongodb/MongoDbDatasourceDialogTest.java

Checkstyle violations:
======================
No errors found

License header violations:
==========================
1. src/main/java/org/pentaho/di/trans/steps/mongodb/MongoDbMeta.java
2. src/main/java/org/pentaho/mongo/wrapper/KettleMongoUtilLogger.java
3. src/main/java/org/pentaho/mongo/wrapper/MongoWrapperUtil.java
4. src/main/java/org/pentaho/mongo/wrapper/field/MongoArrayExpansion.java
5. src/main/java/org/pentaho/mongo/wrapper/field/MongodbInputDiscoverFieldsImpl.java
6. src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputDataTest.java
7. src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputMetaTest.java
8. src/test/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputMetaTest.java
9. src/test/java/org/pentaho/mongo/wrapper/MongoWrapperUtilTest.java







Unit test coverage change:
============
org.pentaho.mongo.wrapper.KettleMongoUtilLogger:
 * branchPercentCoverage :50.00%
 * instructionPercentCoverageChange : +83.67%
 * linePercentCoverageChange : +77.78%
 * complexityPercentCoverageChange : +45.45%
 * methodPercentCoverageChange : +83.33%
    
org.pentaho.mongo.wrapper.MongoWrapperUtil:
 * instructionPercentCoverageChange : +5.36%
 * linePercentCoverageChange : +2.86%
 * complexityPercentCoverageChange : +7.69%
 * methodPercentCoverageChange : +12.50%
    
org.pentaho.di.trans.steps.mongodboutput.MongoDbOutputMeta:
 * instructionPercentCoverageChange : +0.19%
 * branchPercentCoverageChange : +0.93%
 * linePercentCoverageChange : +0.35%
 * complexityPercentCoverageChange : +1.16%
    
org.pentaho.di.ui.trans.steps.mongodbinput.MongoDbInputDialog:
 * linePercentCoverageChange : +0.00%
    
org.pentaho.mongo.wrapper.field.MongodbInputDiscoverFieldsImpl.new MongoDBAction() {...}:
 * branchPercentCoverage :62.50%
 * linePercentCoverage :83.33%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :42.86%
 * instructionPercentCoverage :75.00%
    
org.pentaho.mongo.wrapper.field.MongodbInputDiscoverFieldsImpl:
 * branchPercentCoverage :75.00%
 * linePercentCoverage :90.87%
 * methodPercentCoverage :92.31%
 * complexityPercentCoverage :59.15%
 * instructionPercentCoverage :90.07%
    
org.pentaho.di.trans.steps.mongodboutput.MongoDbOutput:
 * instructionPercentCoverageChange : +57.86%
 * branchPercentCoverageChange : +49.42%
 * linePercentCoverageChange : +62.75%
 * complexityPercentCoverageChange : +27.83%
 * methodPercentCoverageChange : +75.00%
    
org.pentaho.di.trans.steps.mongodbinput.MongoDbInputDiscoverFieldsHolder:
 * branchPercentCoverage :71.43%
 * instructionPercentCoverageChange : +81.88%
 * linePercentCoverageChange : +90.00%
 * complexityPercentCoverageChange : +61.54%
 * methodPercentCoverageChange : +83.33%
    
org.pentaho.di.trans.steps.mongodboutput.MongoDbOutputData:
 * instructionPercentCoverageChange : +8.80%
 * branchPercentCoverageChange : +4.93%
 * linePercentCoverageChange : +11.52%
 * complexityPercentCoverageChange : +11.11%
 * methodPercentCoverageChange : +40.00%
    
org.pentaho.mongo.wrapper.field.MongoField:
 * instructionPercentCoverageChange : +45.22%
 * branchPercentCoverageChange : +46.59%
 * linePercentCoverageChange : +44.36%
 * complexityPercentCoverageChange : +37.93%
 * methodPercentCoverageChange : +27.27%
    
org.pentaho.di.trans.steps.mongodboutput.MongoDbOutputMeta.MongoField:
 * instructionPercentCoverageChange : +28.81%
 * linePercentCoverageChange : +32.14%
 * complexityPercentCoverageChange : +11.11%
 * methodPercentCoverageChange : +25.00%
    
org.pentaho.di.trans.steps.mongodbinput.MongoDbInputData:
 * instructionPercentCoverageChange : +39.42%
 * branchPercentCoverageChange : +39.96%
 * linePercentCoverageChange : +38.53%
 * complexityPercentCoverageChange : +32.43%
 * methodPercentCoverageChange : +20.00%
    
org.pentaho.di.trans.steps.mongodbinput.MongoDbInput:
 * branchPercentCoverage :61.63%
 * linePercentCoverage :85.25%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :38.78%
 * instructionPercentCoverage :82.68%

**Comment by lucboudreau** [comment]
Some license headers are missing, but everything else looks good.

**Comment by mkambol** [comment]
@lucboudreau Missing headers added.

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. mvn -DskipTests -f pom.xml clean install && mvn -f pom.xml site

Cleanup:
-------
mvn build-helper:remove-project-artifact

Changed files:
==============
  pom.xml
  src/main/java/org/pentaho/di/trans/steps/mongodb/MongoDbMeta.java
  src/main/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInput.java
  src/main/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputData.java
  src/main/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputMeta.java
  src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutput.java
  src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputData.java
  src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputDialog.java
  src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputMeta.java
  src/main/java/org/pentaho/di/ui/swing/preview/PreviewRowsSwingDialog.java
  src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/MongoDbInputDialog.java
  src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/MongoDbInputXulDialog.java
  src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/models/MongoDbModel.java
  src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/models/MongoDocumentField.java
  src/main/java/org/pentaho/di/ui/trans/steps/mongodbinput/models/MongoTag.java
  src/main/java/org/pentaho/mongo/wrapper/KettleMongoUtilLogger.java
  src/main/java/org/pentaho/mongo/wrapper/MongoWrapperUtil.java
  src/main/java/org/pentaho/mongo/wrapper/field/MongoArrayExpansion.java
  src/main/java/org/pentaho/mongo/wrapper/field/MongoField.java
  src/main/java/org/pentaho/mongo/wrapper/field/MongodbInputDiscoverFieldsImpl.java
  src/test/java/org/pentaho/di/trans/steps/mongodbinput/BaseMongoDbStepTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputDataTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputDiscoverFieldsHolderTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputMetaTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodbinput/MongoDbInputTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputDataTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputMetaTest.java
  src/test/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputTest.java
  src/test/java/org/pentaho/mongo/wrapper/KettleMongoUtilLoggerTest.java
  src/test/java/org/pentaho/mongo/wrapper/MongoFieldTest.java
  src/test/java/org/pentaho/mongo/wrapper/MongoWrapperUtilTest.java
  src/test/java/org/pentaho/mongo/wrapper/field/MongodbInputDiscoverFieldsImplTest.java
  src/test/java/org/pentaho/reporting/ui/datasources/mongodb/MongoDbDatasourceDialogTest.java

Checkstyle violations:
======================
No errors found

License header violations:
==========================
No violations found








Unit test coverage change:
============
org.pentaho.mongo.wrapper.KettleMongoUtilLogger:
 * branchPercentCoverage :50.00%
 * instructionPercentCoverageChange : +83.67%
 * linePercentCoverageChange : +77.78%
 * complexityPercentCoverageChange : +45.45%
 * methodPercentCoverageChange : +83.33%
    
org.pentaho.mongo.wrapper.MongoWrapperUtil:
 * instructionPercentCoverageChange : +5.36%
 * linePercentCoverageChange : +2.86%
 * complexityPercentCoverageChange : +7.69%
 * methodPercentCoverageChange : +12.50%
    
org.pentaho.di.ui.trans.steps.mongodbinput.MongoDbInputDialog:
 * linePercentCoverageChange : +0.00%
    
org.pentaho.mongo.wrapper.field.MongodbInputDiscoverFieldsImpl.new MongoDBAction() {...}:
 * branchPercentCoverage :62.50%
 * linePercentCoverage :83.33%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :42.86%
 * instructionPercentCoverage :75.00%
    
org.pentaho.mongo.wrapper.field.MongodbInputDiscoverFieldsImpl:
 * branchPercentCoverage :75.00%
 * linePercentCoverage :90.87%
 * methodPercentCoverage :92.31%
 * complexityPercentCoverage :59.15%
 * instructionPercentCoverage :90.07%
    
org.pentaho.di.trans.steps.mongodboutput.MongoDbOutput:
 * instructionPercentCoverageChange : +57.86%
 * branchPercentCoverageChange : +49.42%
 * linePercentCoverageChange : +62.75%
 * complexityPercentCoverageChange : +27.83%
 * methodPercentCoverageChange : +75.00%
    
org.pentaho.di.trans.steps.mongodbinput.MongoDbInputDiscoverFieldsHolder:
 * branchPercentCoverage :71.43%
 * instructionPercentCoverageChange : +81.88%
 * linePercentCoverageChange : +90.00%
 * complexityPercentCoverageChange : +61.54%
 * methodPercentCoverageChange : +83.33%
    
org.pentaho.di.trans.steps.mongodboutput.MongoDbOutputData:
 * instructionPercentCoverageChange : +8.80%
 * branchPercentCoverageChange : +4.93%
 * linePercentCoverageChange : +11.52%
 * complexityPercentCoverageChange : +11.11%
 * methodPercentCoverageChange : +40.00%
    
org.pentaho.mongo.wrapper.field.MongoField:
 * instructionPercentCoverageChange : +45.22%
 * branchPercentCoverageChange : +46.59%
 * linePercentCoverageChange : +44.36%
 * complexityPercentCoverageChange : +37.93%
 * methodPercentCoverageChange : +27.27%
    
org.pentaho.di.trans.steps.mongodboutput.MongoDbOutputMeta.MongoField:
 * instructionPercentCoverageChange : +28.81%
 * linePercentCoverageChange : +32.14%
 * complexityPercentCoverageChange : +11.11%
 * methodPercentCoverageChange : +25.00%
    
org.pentaho.di.trans.steps.mongodbinput.MongoDbInputData:
 * instructionPercentCoverageChange : +39.42%
 * branchPercentCoverageChange : +39.96%
 * linePercentCoverageChange : +38.53%
 * complexityPercentCoverageChange : +32.43%
 * methodPercentCoverageChange : +20.00%
    
org.pentaho.di.trans.steps.mongodbinput.MongoDbInput:
 * branchPercentCoverage :61.63%
 * linePercentCoverage :85.25%
 * methodPercentCoverage :100.00%
 * complexityPercentCoverage :38.78%
 * instructionPercentCoverage :82.68%

File: src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputData.java [hunk 1]
```diff
@@ -958,44 +945,4 @@ protected static MongoTopLevel checkTopLevelConsistency( List<MongoDbOutputMeta.
     return MongoTopLevel.ARRAY;
   }
 
-  public static void main( String[] args ) {
```
Comment (lucboudreau) [inline src/main/java/org/pentaho/di/trans/steps/mongodboutput/MongoDbOutputData.java]: :)

**Pull request merged**